src/text/non_blocking/raw_text_reader.rs

```diff
@@ -404,9 +404,11 @@ impl<A: AsRef<[u8]> + Expandable> RawTextReader<A> {
     }
 
     /// Assumes that the reader is inside a struct AND that a field has already been successfully
-    /// parsed from input using [next_struct_field_name] and attempts to parse the next value.
+    /// parsed from input using [`next_struct_field_name`][1] and attempts to parse the next value.
     /// In this input position, only a value (or whitespace/comments) are legal. Anything else
     /// (including EOF) will result in a decoding error.
+    ///
+    /// [1]: Self::next_struct_field_name
```

Comment:
Only two examples in this patch of a reference-style link, and none that I can find outside it. Might as well stick with inline links, and if we find a more complicated case use named reference-style links. 
```suggestion
    /// parsed from input using [`next_struct_field_name`](Self::next_struct_field_name) and
    /// attempts to parse the next value.
    /// In this input position, only a value (or whitespace/comments) are legal. Anything else
    /// (including EOF) will result in a decoding error.
```